### PR TITLE
Endpoint to get job runs by job

### PR DESCRIPTION
**Pull request by sshah-wework**
New endpoint to fetch job runs by job:
- Returns job runs in reverse chronological order
- Supports `LIMIT` and `OFFSET` for pagination

### CODE REVIEW

File: src/main/java/marquez/api/resources/JobResource.java [hunk 1]
```diff
@@ -149,12 +151,25 @@ public Response create(
...
+    if (!namespaceService.exists(namespaceName)) {
+      return Response.status(Response.Status.NOT_FOUND).entity("Namespace not found").build();
+    }
+    final List<JobRun> jobRuns = jobService.getAllRunsOfJob(namespaceName, job, limit, offset);
+    return Response.ok().entity(JobRunResponseMapper.map(jobRuns)).build();
+  }
+
+  @GET
+  @Timed
+  @Path("/jobs/runs/{id}")
```
Comment (wslulciuc) [inline src/main/java/marquez/api/resources/JobResource.java]: _Minor_: We'll want to keep the metric annotations for `GET` `/jobs/runs/{id}`. May have been removed when merging with `master`:

```java
@Timed
@ResponseMetered
@ExceptionMetered
@GET
@Path("/jobs/runs/{id}")
@Produces(APPLICATION_JSON)
public Response get(@PathParam("id") final UUID runId) throws MarquezServiceException {
  . . .
}
``

File: src/main/java/marquez/api/resources/JobResource.java [hunk 2]
```diff
@@ -149,12 +151,25 @@ public Response create(
         .build();
   }
 
-  @Timed
-  @ResponseMetered
-  @ExceptionMetered
   @GET
-  @Path("/jobs/runs/{id}")
   @Produces(APPLICATION_JSON)
+  @Path("/namespaces/{namespace}/jobs/{job}/runs")
```
Comment (wslulciuc) [inline src/main/java/marquez/api/resources/JobResource.java]: We'll also want to add the following metric annotations:

```java
@Timed
@ResponseMetered
@ExceptionMetered
@GET
@Path("/namespaces/{namespace}/jobs/{job}/runs")
@Produces(APPLICATION_JSON)
public Response getRunsForJob(...) {
  ...
}
```

File: src/main/java/marquez/db/JobRunDao.java [hunk 3]
```diff
@@ -68,4 +69,15 @@ default void updateState(UUID jobRunID, Integer state) {
...
           + "WHERE jr.guid = :guid")
   JobRun findJobRunById(UUID guid);
+
+  @SqlQuery(
+      "SELECT jr.*, jra.args_json "
+          + "FROM job_runs jr "
+          + "INNER JOIN job_versions jv "
+          + " ON (jr.job_version_guid = jv.guid AND jv.job_guid = :jobUuid)"
+          + "LEFT JOIN job_run_args jra "
+          + " ON (jr.job_run_args_hex_digest = jra.hex_digest) "
```
Comment (wslulciuc) [inline src/main/java/marquez/db/JobRunDao.java]: _Minor_: Just realized we need to update the table schemas for `job_runs` and `job_runs_args` (which drops `job_run_args_hex_digest`, etc)

- Update job_run_args schema #541
- Update job_runs schema #542

File: src/main/java/marquez/db/JobRunDao.java [hunk 4]
```diff
@@ -68,4 +69,15 @@ default void updateState(UUID jobRunID, Integer state) {
...
+          + "ORDER BY created_at DESC "
+          + "LIMIT :limit OFFSET :offset")
```
Comment (wslulciuc) [inline src/main/java/marquez/db/JobRunDao.java]: 👍

File: src/test/java/marquez/api/resources/JobResourceTest.java [hunk 5]
```diff
@@ -322,6 +326,69 @@ public void testJobRunCompleteWithUnparseableRunId() throws Exception {
...
+
+    Response response =
+        JOB_RESOURCE.getRunsForJob(
+            NAMESPACE_NAME, "nonexistent_job", DEFAULT_LIMIT, DEFAULT_OFFSET);
+
+    assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
+  }
+
+  @Test
+  public void testGetAllRunsOfJob_noJobRuns() throws MarquezServiceException {
```
Comment (wslulciuc) [inline src/test/java/marquez/api/resources/JobResourceTest.java]: If no job runs are found for a given job, I think I'd be more ideal to return an empty set of runs rather then a `404`?

```
{
  "runs": []
}
```

File: src/test/java/marquez/api/resources/JobResourceTest.java [hunk 6]
```diff
@@ -322,6 +326,69 @@ public void testJobRunCompleteWithUnparseableRunId() throws Exception {
...
+    JobResponse job = generateApiJob();
+    when(MOCK_NAMESPACE_SERVICE.exists(NAMESPACE_NAME)).thenReturn(false);
```
Comment (wslulciuc) [inline src/test/java/marquez/api/resources/JobResourceTest.java]: Also, I think we want to this mocked call to be:

```java
when(MOCK_NAMESPACE_SERVICE.exists(NAMESPACE_NAME)).thenReturn(true);
``

File: src/main/java/marquez/db/JobRunDao.java (hunk 3 above)
Comment (sshah-wework) [reply src/main/java/marquez/db/JobRunDao.java]: good reminder, thanks

File: src/test/java/marquez/api/resources/JobResourceTest.java (hunk 5 above)
Comment (sshah-wework) [reply src/test/java/marquez/api/resources/JobResourceTest.java]: fixed

File: src/test/java/marquez/api/resources/JobResourceTest.java (hunk 6 above)
Comment (sshah-wework) [reply src/test/java/marquez/api/resources/JobResourceTest.java]: typo on my part, fixed

File: src/main/java/marquez/api/resources/JobResource.java (hunk 1 above)
Comment (sshah-wework) [reply src/main/java/marquez/api/resources/JobResource.java]: yah was a bad merge, should've checked the diff. fixing it

Comment (sshah-wework) [reply src/main/java/marquez/api/resources/JobResource.java]: fixed

File: src/main/java/marquez/api/resources/JobResource.java (hunk 2 above)
Comment (sshah-wework) [reply src/main/java/marquez/api/resources/JobResource.java]: totally missed this good catch

**Pull request merged**